extensions/azcli/src/azdata.ts

```diff
@@ -77,181 +69,167 @@ export class AzdataTool implements azdataExt.IAzdataApi {
+				const argsArray = ['sql', 'mi-arc', 'edit', '-n', name, '--k8s-namespace', namespace, '--use-k8s'];
+				if (args.coresLimit) { argsArray.push('--cores-limit', args.coresLimit); }
+				if (args.coresRequest) { argsArray.push('--cores-request', args.coresRequest); }
+				if (args.memoryLimit) { argsArray.push('--memory-limit', args.memoryLimit); }
+				if (args.memoryRequest) { argsArray.push('--memory-request', args.memoryRequest); }
+				if (args.noWait) { argsArray.push('--no-wait'); }
+				return this.executeCommand<void>(argsArray, additionalEnvVars);
+			}
 		}
-		return this.executeCommand<void>(args, Object.assign({}, additionalEnvVars, { 'AZDATA_PASSWORD': password }), azdataContext);
-	}
+	};
 
 	/**
-	 * Gets the output of running '--version' command on the azdata tool.
+	 * Gets the output of running '--version' command on the az tool.
 	 * It also updates the cachedVersion property based on the return value from the tool.
 	 */
-	public async version(): Promise<azdataExt.AzdataOutput<string>> {
-		const output = await executeAzdataCommand(`"${this._path}"`, ['--version']);
+	public async version(): Promise<azExt.AzOutput<string>> {
+		const output = await executeAzCommand(`"${this._path}"`, ['--version']);
 		this._semVersion = new SemVer(parseVersion(output.stdout));
 		return {
-			logs: [],
-			stdout: output.stdout.split(os.EOL),
-			stderr: output.stderr.split(os.EOL),
-			result: output.stdout
+			stdout: output.stdout
+			// stderr: output.stderr.split(os.EOL), TODOCANYE
+			// result: output.stdout
 		};
 	}
 
 	/**
-	 * Executes the specified azdata command.
-	 * @param args The args to pass to azdata
+	 * Executes the specified az command.
+	 * @param args The args to pass to az
 	 * @param additionalEnvVars Additional environment variables to set for this execution
 	 */
-	public async executeCommand<R>(args: string[], additionalEnvVars?: azdataExt.AdditionalEnvVars, azdataContext?: string): Promise<azdataExt.AzdataOutput<R>> {
+	public async executeCommand<R>(args: string[], additionalEnvVars?: azExt.AdditionalEnvVars): Promise<azExt.AzOutput<R>> {
 		try {
-			if (azdataContext) {
-				args = args.concat('--controller-context', azdataContext);
-			}
-			const output = JSON.parse((await executeAzdataCommand(`"${this._path}"`, args.concat(['--output', 'json']), additionalEnvVars)).stdout);
+			// JSON OUTPUT
+			const result = await executeAzCommand(`"${this._path}"`, args.concat(['--output', 'json']), additionalEnvVars);
+
+			const output = JSON.parse(result.stdout);
 			return {
-				logs: <string[]>output.log,
-				stdout: <string[]>output.stdout,
-				stderr: <string[]>output.stderr,
-				result: <R>output.result
+				stdout: <R>output
 			};
 		} catch (err) {
 			if (err instanceof ExitCodeError) {
 				try {
-					// For azdata internal errors the output is JSON and so we need to do some extra parsing here
+					// For az internal errors the output is JSON and so we need to do some extra parsing here
```

Comment:
FYI I doubt this will work/be necessary anymore given the different output format. So you should check that at some point. Probably best to just remove it for now though.